src/transformers/models/bigscience176b/scale_mask_softmax.py

```diff
@@ -0,0 +1,88 @@
+# coding=utf-8
+# Copyright (c) 2020, NVIDIA CORPORATION.  All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+import enum
+import torch
+import torch.nn as nn
+
+class AttnMaskType(enum.Enum):
+    padding = 1
+    causal = 2
+    prefix = 3
+
+class ScaleMaskSoftmax(nn.Module):
```

Comment:
I think we could remove this whole file and replace it by a nn.Softmax in the modeling.py file for a couple of reasons:
- in `transformers` we strongly favor trying to have the whole forward pass in a file file if possible
- keep in mind that this is a pretrained model that will be provided "as is". As such we don't expect people to reuse this class for anything else or to tweak the arguments. In other words can hardcode as much as you want.

Reply:
Perfect! Thanks for the explanation, fixed everything in [this commit](https://github.com/younesbelkada/transformers/commit/adcfc157b9dabf40cffd0a1b5b685438726b03f8), no need to add an extra file anymore now :100: